src/ImageSharp/ColorProfiles/ColorProfileConverterExtensionsIcc.cs

```diff
@@ -33,64 +34,103 @@ internal static TTo ConvertUsingIccProfile<TFrom, TTo>(this ColorProfileConverte
             MemoryAllocator = converter.Options.MemoryAllocator,
 
             // TODO: Double check this but I think these are normalized values.
-            SourceWhitePoint = CieXyz.FromScaledVector4(new(converter.Options.SourceIccProfile.Header.PcsIlluminant, 1F)),
-            TargetWhitePoint = CieXyz.FromScaledVector4(new(converter.Options.TargetIccProfile.Header.PcsIlluminant, 1F)),
+            SourceWhitePoint = new CieXyz(converter.Options.SourceIccProfile.Header.PcsIlluminant),
+            TargetWhitePoint = new CieXyz(converter.Options.TargetIccProfile.Header.PcsIlluminant),
         });
 
         IccDataToPcsConverter sourceConverter = new(converter.Options.SourceIccProfile);
         IccPcsToDataConverter targetConverter = new(converter.Options.TargetIccProfile);
-        IccColorSpaceType sourcePcsType = converter.Options.SourceIccProfile.Header.ProfileConnectionSpace;
-        IccColorSpaceType targetPcsType = converter.Options.TargetIccProfile.Header.ProfileConnectionSpace;
-        IccVersion sourceVersion = converter.Options.SourceIccProfile.Header.Version;
-        IccVersion targetVersion = converter.Options.TargetIccProfile.Header.Version;
-
-        Vector4 pcs = sourceConverter.Calculate(source.ToScaledVector4());
-
-        // Profile connecting spaces can only be Lab, XYZ.
-        if (sourcePcsType is IccColorSpaceType.CieLab && targetPcsType is IccColorSpaceType.CieXyz)
-        {
-            // Convert from Lab to XYZ.
-            CieLab lab = CieLab.FromScaledVector4(pcs);
-            CieXyz xyz = pcsConverter.Convert<CieLab, CieXyz>(in lab);
-            pcs = xyz.ToScaledVector4();
-        }
-        else if (sourcePcsType is IccColorSpaceType.CieXyz && targetPcsType is IccColorSpaceType.CieLab)
+        IccProfileHeader sourceHeader = converter.Options.SourceIccProfile.Header;
+        IccProfileHeader targetHeader = converter.Options.TargetIccProfile.Header;
+        IccColorSpaceType sourcePcsType = sourceHeader.ProfileConnectionSpace;
+        IccColorSpaceType targetPcsType = targetHeader.ProfileConnectionSpace;
+        IccRenderingIntent sourceIntent = sourceHeader.RenderingIntent;
+        IccRenderingIntent targetIntent = targetHeader.RenderingIntent;
+        IccVersion sourceVersion = sourceHeader.Version;
+        IccVersion targetVersion = targetHeader.Version;
+
+        // all conversions are funnelled through XYZ in case PCS adjustments need to be made
+        CieXyz xyz;
```

Comment:
I'll take a shot at avoiding the overhead when unnecessary. I expect it will result in functions that look very similar like `ConvertIcc()` and `ConvertIccWithPerceptualAdjustment()` - I can't see a natural `if (adjustmentNeeded) { PerformExtraStep(); }` at the moment